lom/src/lib/lomipc/server_transport.go

```diff
@@ -390,18 +402,55 @@ func GetLoMTransport() *LoMTransport {
 }
 
 /* Init the serverside transport */
-func ServerInit() (*LoMTransport, error) {
-    tr := GetLoMTransport()
+func httpServerInit() error {
 
-    rpc.Register(tr)
     rpc.HandleHTTP()
-    l, e := net.Listen("tcp", ":1234")
+    l, e := net.Listen("tcp", "localhost:"+strconv.Itoa(RPC_HTTP_PORT))
     if e != nil {
-        LogPanic("listen error:(%v)", e)
-        return nil, e
+        LogPanic("listen error at %d :(%v)", RPC_HTTP_PORT, e)
+        return e
     }
     go http.Serve(l, nil)
-    LogDebug("Server: Started serving")
+    LogDebug("HTTP Server: Started serving")
+
+    return nil
+}
+
+func jsonServerInit() error {
```

Comment:
should we close the connections while exiting ? l.close() and conn.close() ??

Reply:
If listen fails, nothing to close.

There after it is in the forever loop of accept. Successfully accepted are served. There is nothing to close.

Reply:
closing listen when function returns.

Reply:
The embedded go fn runs forever looking for new connection until engine exits. This fn uses the listener